modules/finance/src/main/java/com/opengamma/strata/finance/rate/bond/FixedCouponBond.java

```diff
@@ -0,0 +1,784 @@
+/**
+ * Copyright (C) 2015 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.finance.rate.bond;
+
+import java.io.Serializable;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Set;
+
+import org.joda.beans.Bean;
+import org.joda.beans.BeanDefinition;
+import org.joda.beans.ImmutableBean;
+import org.joda.beans.JodaBeanUtils;
+import org.joda.beans.MetaProperty;
+import org.joda.beans.Property;
+import org.joda.beans.PropertyDefinition;
+import org.joda.beans.impl.direct.DirectFieldsBeanBuilder;
+import org.joda.beans.impl.direct.DirectMetaBean;
+import org.joda.beans.impl.direct.DirectMetaProperty;
+import org.joda.beans.impl.direct.DirectMetaPropertyMap;
+
+import com.google.common.collect.ImmutableList;
+import com.opengamma.strata.basics.currency.CurrencyAmount;
+import com.opengamma.strata.basics.currency.Payment;
+import com.opengamma.strata.basics.date.DayCount;
+import com.opengamma.strata.basics.date.DaysAdjustment;
+import com.opengamma.strata.basics.schedule.PeriodicSchedule;
+import com.opengamma.strata.basics.schedule.Schedule;
+import com.opengamma.strata.basics.schedule.SchedulePeriod;
+import com.opengamma.strata.collect.id.StandardId;
+
+/**
+ * A fixed coupon bond.
+ * <p>
+ * A fixed coupon bond is a financial instrument that represents a stream of fixed payments. 
+ * The payments consist two types: periodic coupon payments and nominal payment.
+ * The periodic payments are made {@code n} times a year with a fixed coupon rate at individual coupon dates.   
+ * The nominal payment is the unique payment at the final coupon date.
+ * <p>
+ * The periodic coupon payment schedule is defined using {@link PeriodicSchedule}. 
+ * The payment amount is computed with {@code fixedRate} and {@code notionalAmount}. 
+ * The nominal payment is defined from the last period of the periodic coupon payment schedule and {@code notionalAmount}. 
+ * <p>
+ * The accrual factor between two dates is computed {@code dayCount}. 
+ * The legal entity of this fixed coupon bond is identified by {@link StandardId}. The enum, {@link YieldConvention}, 
+ * specifies the yield computation convention, and {@link DaysAdjustment} does the number of days between valuation 
+ * date and settlement date. 
+ * </ul>
+ */
+@BeanDefinition
+public final class FixedCouponBond
+    implements FixedCouponBondProduct, ImmutableBean, Serializable {
+
+  /**
+   * The notional amount of the product.
+   * <p>
+   * The amount is signed.
+   * A positive amount indicates the payments are to be received.
+   * A negative amount indicates the payments are to be paid.
```

Comment:
The notional should always be positive. Here we describe a fungible bond. The quantity will appear in the trade.